src/features/auth/auth.js

```diff
@@ -1,27 +1,96 @@
 import React from 'react';
+import { useSelector, shallowEqual } from 'react-redux';
+import { Link, useHistory } from 'react-router-dom';
+import { Formik, Field, Form } from 'formik';
+import * as yup from 'yup';
+import classNames from 'classnames';
+
+import { Button, WithLoading } from '../../components/index.js';
+import { useActions } from '../../shared/index.js';
+import { login, currentUserSelector } from '../../models/index.js';
 import styles from './auth.scss';
 
-export const Auth = () => (
-  <div className={styles.wrapper}>
-    <div className="container">
-      <div className="row justify-content-center">
-        <div className="col-lg-4 col-md-6 col-sm-8">
-          <div className={styles.form}>
-            <form className="p-3">
-              <div className="form-group">
-                <input type="email" className="form-control" id="auth-form__email" placeholder="Email address" />
-              </div>
-              <div className="form-group">
-                <input type="password" className="form-control" id="auth-form__password" placeholder="Password" />
-              </div>
-              <button type="submit" className="btn btn-block btn-warning">Sign in</button>
-              <div className="text-center mt-3">
-                <p>Don't have an account? <a href="#" className={styles.link}>Registration</a></p>
-              </div>
-            </form>
+export const Auth = () => {
+  const {
+    isLoading,
+    error: requestError,
+    loaded,
+  } = useSelector(currentUserSelector, shallowEqual);
+  const dispatchLogIn = useActions(login);
+  const history = useHistory();
+
+  const submitHandler = (values) => {
+    dispatchLogIn(values);
+  };
+
+  if (loaded && !requestError) {
+    history.push('/home');
+  }
+
+  const validationSchema = yup.object().shape({
+    email: yup.string()
```

Comment:
I think we need to write all functions for validation, like this one, in validation-helpers.js